plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -90,6 +96,38 @@ public InstanceList run() throws Exception {
         return instances;
     }
 
+    @Override
+    public String networkAddress(String gceName) throws IOException {
+        String urlMetadataNetwork = GCE_METADATA_URL + "/" + gceName;
+        logger.debug("get network information for [{}]", gceName);
+        URL url = new URL(urlMetadataNetwork);
+        HttpHeaders headers;
+        try {
+            // hack around code messiness in GCE code
+            // TODO: get this fixed
+            headers = AccessController.doPrivileged(new PrivilegedExceptionAction<HttpHeaders>() {
+                @Override
+                public HttpHeaders run() throws IOException {
+                    return new HttpHeaders();
+                }
+            });
+
+            // This is needed to query meta data: https://cloud.google.com/compute/docs/metadata
+            headers.put("Metadata-Flavor", "Google");
+            HttpResponse response;
+            response = getGceHttpTransport().createRequestFactory()
+                    .buildGetRequest(new GenericUrl(url))
+                    .setHeaders(headers)
+                    .execute();
+            String ip = response.parseAsString();
+            logger.debug("ip found [{}]", ip);
+            return ip;
+        } catch (Exception e) {
+            logger.warn("unable to create GCE Http Client", e);
```

Comment:
I think there is something wrong with GCE http client code and we will need this around it, we grant the permission as a workaround. It would be really good if we could get this fixed in their code though. It makes it difficult for apps to protect credentials etc to their services!